tenpy/tools/spectral_function_tools.py

```diff
@@ -0,0 +1,258 @@
+        # if k is indexed like (kx, ky) a coordinate (2d) is returned.
+        k = k_x[:, np.newaxis, :] + k_y[np.newaxis, :, :]
+    return ft_space, k
+
+
+def fourier_transform_time(a, dt, axis=0):
+    # fourier transform in time
+    # (note that ifft is used, resulting in a minus sign in the exponential)
+    ft_time = np.fft.ifft(a, axis=axis) * a.shape[axis]  # renormalize
+    w = np.fft.fftfreq(len(ft_time), dt / (2 * np.pi))
+    # shifting
+    ft_time = np.fft.fftshift(ft_time, axes=axis)
+    w = np.fft.fftshift(w)
+    return ft_time, w
+
+
+def apply_gaussian_windowing(a, sigma: float = 0.4, axis=0):
+    """Simple gaussian windowing function along an axes.
+
+    Applying a windowing function avoids Gibbs oscillation. tn are time steps 0, 1, ..., N
+
+    Parameters
+    ----------
+    a : ndarray
+        a ndarray where the time series is along axis `axes`
+    sigma : float
+        standard-deviation used for the gaussian window
+    axis : int
+        axes along which to apply the gaussian window
+
+    Returns
+    -------
+    np.ndarray
+    """
+    # extract number of time-steps
+    n_tsteps = a.shape[axis]
+    tn = np.arange(n_tsteps)
+    # create gaussian windowing function with the right length
+    gaussian_window = np.exp(-0.5 * (tn / (n_tsteps * sigma))**2)
+    # swap dimension which should be weighted (window applied to) to last dim (for np broadcasting)
+    swapped_a = np.swapaxes(a, -1, axis)
+    weighted_arr = swapped_a * gaussian_window  # apply window
+    return np.swapaxes(weighted_arr, axis, -1)
+
+
+def to_mps_geometry(lat, a):
+    """Bring measurement in lattice geometry to mps geometry.
+
+    This assumes that the array a has shape (..., Lx, Ly, Lu),
+    or if Lu = 1, (..., Lx, Ly)
+    """
+    mps_idx_flattened = np.ravel_multi_index(tuple(lat.order.T), lat.shape)
+    dims_until_lat_dims = a.ndim - (lat.dim + 1)  # add unit cell dim
+    if lat.Lu == 1:
+        dims_until_lat_dims += 1
+    a = a.reshape(a.shape[:dims_until_lat_dims] + (-1, ))
+    a = np.take(a, mps_idx_flattened, axis=-1)
+    return a
+
+
+def plot_correlations_on_lattice(ax,
+                                 lat,
+                                 correlations,
+                                 pairs='nearest_neighbors',
+                                 scale=1,
+                                 color_pos='r',
+                                 color_neg='g',
+                                 color=None,
+                                 zorder=0):
+    """Function to plot correlations on a lattice.
+
+    The strength of the correlations is given by the thickness of connecting lines.
+
+    Parameters
+    ----------
+    ax :
+        `matplotlib.axes.Axes`
+    lat :
+        a (TeNPy) lattice to plot the correlations on
+    correlations : array-like
+        an array of correlations (in mps_form)
+    pairs: str
+    scale: float
+    color_pos: str
+    color_neg: str
+    color: str
+    zorder: float
```

Comment:
docs. 

It might also be nice to have an alternative version that has fixed line width, but changes color according to a colormap (e.g. red-blue if we have positive/negative values). That would probably be a different function, though.